### ISSUE TITLE
UID field can't remain the same when using i18n

### ISSUE THREAD

**Issue by kavalcante** [strapi/strapi#10107]
### Describe the bug

When adding i18n plugin and creating or editing a content in another language. The UID field, generates another value instead the original one. 

### Steps to reproduce the behavior

1. Create an collection called `Posts` or whatever.
2. Create an UID field called `slug`
3. Add one or more locations in settings.
4. Create the default version with the slug `test` at english version, for example.
5. Save and then go to the other language. In my case, I'm going with pt_BR. After changing for the location, the slug generate is other than `test`, probably will be `test-1`.

### Expected behavior

The UID should only block unique UID from others content, not the translations one.

### Screenshots

![image](https://user-images.githubusercontent.com/13931503/115769611-1ba7f480-a382-11eb-8fa4-e5aeacf042d4.png)

### System

- Node.js version: v14 
- NPM version: 6.14.11
- Strapi version: 3.6.0
- Database: SQL Lite 5.0.0
- Operating system: Windows 10

**Comment by AlexStansfield** [comment]
I was also expecting it to work this way. Maybe when configuring the UID there can be an option to combine its uniqueness against the locale?

I think i want to do the same as the OP, have a single slug in multiple locales.

I had already worked around this previously by having a "pages" collection and a "page-slugs" collection. 

The "page-slug" record holds the unique slug field and then a relation to pages for each locale. 
![Screen Shot 2564-04-23 at 15 09 19](https://user-images.githubusercontent.com/4319443/115840458-e8c73480-a445-11eb-8b6f-ec33268e02ec.png)

Was hoping this new i18n feature would see the end of it :D

**Comment by alexverdaguer** [comment]
Having same issue here. The only workaround is using a text field as slug, but filling this field manually, not ideal so far.

**Comment by kavalcante** [comment]
I understand that in a blog post, is better to have a translated slug. But in my case I'm adding something that has a unique name, therefore will not be translated.

**Comment by alexandreec** [comment]
Having a translated slug it could be a solution, but only if you can define that slug only available for that translation, if not, filling the form is a mess, you will have to be dealing with slug suggestions always.

**Comment by alexandreec** [comment]
But keep in mind that an UUID it has to be unique for definition, we will have to convert the slugs to normal inputs and lose the auto-generation button.

Another solution is to put the current locale in the slug by default instead of adding  "-1":
"page-slug"
"page-slug-ca" (Catalan)
"page-slug-es" (Spanish)

**Comment by alexandreec** [comment]
But keep in mind that an UUID it has to be unique by definition, we will have to convert the slugs to normal inputs and lose the autogeneration button.

Another solution is to put the current locale in the slug by default instead of adding  "-1":
"page-slug"
"page-slug-ca" (Catalan)
"page-slug-es" (Spanish)

**Comment by alexandreec** [comment]
But keep in mind that an UUID it has to be unique by definition, we will have to convert the slugs to normal inputs and lose the auto-generation button.

Another solution is to put the current locale always in the slug by default:
"page-slug"
"page-slug.-ca" (Catalan)
"page-slug-es" (Spanish)

Then you know that if you are not in your default locale, you will have to add "-locale_code" to the queried slug to obtain the data from Strapi.

**Comment by alexandreec** [comment]
But keep in mind that an UUID it has to be unique by definition, we will have to convert the slugs to normal inputs and lose the auto-generation button.

Another solution is to put the current locale always in the slug by default:
"page-slug"
"page-slug-ca" (Catalan)
"page-slug-es" (Spanish)

Then you know that if you are not in your default locale, you will have to add "-locale_code" to the queried slug to obtain the data from Strapi.

**Comment by alexandreec** [comment]
But keep in mind that an UID it has to be unique by definition, we will have to convert the slugs to normal inputs and lose the auto-generation button.

Another solution is to put the current locale always in the slug by default:
"page-slug"
"page-slug-ca" (Catalan)
"page-slug-es" (Spanish)

Then you know that if you are not in your default locale, you will have to add "-locale_code" to the queried slug to obtain the data from Strapi.

**Comment by alexandreec** [comment]
But keep in mind that an UID it has to be unique by definition, we will have to convert the slugs to normal inputs and lose the auto-generation button, but this is not an elegant solution because you are messing inputs with locales.

**Comment by alexandreec** [comment]
A solution is to put the other locales in the slug:

"page-slug" 
"page-slug-ca" (Catalan) 

Then you know that if you are not in your default locale, you will have to add "-locale_code" to the queried slug to obtain the data from Strapi.

**Comment by alexandreec** [comment]
A solution is to put the other locales in the slug:

"page-slug"
"page-slug--ca" (Catalan)

Then you know that if you are not in your default locale, you will have to add "--locale_code" to the queried slug to obtain the data from Strapi.

**Comment by kavalcante** [comment]
@alexandreec I understand that but in my opinion, I'm not creating a new content. I am just translating an existing one. If this is not possible because of the architeture or something like this, ok. I will need to find another way, probably what is described here: [https://strapi.io/documentation/developer-docs/latest/guides/slug.html#create-a-slug-system](https://strapi.io/documentation/developer-docs/latest/guides/slug.html#create-a-slug-system)

**Comment by michael-proulx** [comment]
I'm looking into that feature too, need it for our blog, like the authors need to use the same uid.

ATM I'm doing a tricky way, by deleting the slug indexes and create my custom one.
In mongoDB we can use compond.
![image](https://user-images.githubusercontent.com/23478544/116117752-97f44d80-a68a-11eb-812e-6b3c1e8f0755.png)

**Comment by alexandreec** [comment]
@kavalcante You are right, is not possible because i18n only creates a reference to an entry in the same table, so you can only have one UID. The way that is described in your link seams the most reliable solution and works well, I disabled the input in the views configuration and created a unique key along with the locale field for better query performance. Thank you for the suggestion ;)

**Comment by robinComa** [comment]
Same issue for me, I would like the same slug for my content in FR and EN because I want to be able to switch from FR lang to EN lag without request slug to generate the new URL => **fr/article/:slug => en/blog/:slug**

For the moment the UID type cannot be outside the localisation, so I try to deal with it, and I have another issue with the workaround (use diffrents slug for all langs) : fr/article/:fr-slug => fr/article/:en-slug

I have a custom controller like this : 
const entity = await strapi.services.blog.findOne({ slug, locale });

Here exemples of diffrents use case : 

- strapi.services.blog.findOne(**{ slug: 'fr-slug', locale: 'fr' }**) => **OK** (FR content response)
- strapi.services.blog.findOne(**{ slug: 'en-slug', locale: 'en' }**) => **KO** (empty response)
- strapi.services.blog.findOne(**{ slug: 'fr-slug', locale: 'en' }**) => **KO** (FR content response)

Any idea ?

**Comment by sam-pires** [comment]
> Hey @sam-pires, can you share your vision as PM about this? There's any intend to develop on this?

Hi @kavalcante,
So as mentioned in the discussions above, UIDs are indeed unique by definition. It's a limitation of our current model, we can't have the 2 same UIDs, even if they are on different locales of the same entity. It's because 2 locales of an entity are actually 2 distinct entities.
That's also why localization is forced on UIDs (you can't decide to localize or not UIDs).

That's not something we can change in the v3, but that's something we'll solve with the new data model in the v4. That's definitely something we want to allow with i18n (and in other situations as well, e.g. with an improved Draft&Publish feature).

As @alexverdaguer mentioned above, as a workaround you can use a text field and deactivate i18n on this particular field, so it's the same on all locales.

Hope it helps 🙂

**Comment by derrickmehaffy** [comment]
> I'm another developer bitten by this. Really expected UID to provide something like a "scope".
> 
> @sam-pires what's the Progress on v4 and this situation?

v4 is currently in public beta and a rough timeline for stable is in Nov (no exact date yet)
Regarding this in v4 though, I'll need to loop in @JabStrapi or @alexandrebodin 
(Sam is no longer working at Strapi)

**Comment by petercherm** [comment]
I've hit the same problem. I need to use some unique IDs which need to stay the same across the localised versions of the entry. Is this issue being addressed?

**Comment by alexandrebodin** [comment]
This should be considered as a feature request. We did not change the way the feature worked with i18n with the v4 as we focused on technical topics. I'll change it to feature request so this can be considered by the team :)

**Comment by marcospassos** [comment]
Same problem here :/

**Comment by derrickmehaffy** [comment]
Closing and locking in favor of the tracking issue: https://github.com/strapi/strapi/issues/12538

**Comment by jhoward1994** [comment]
This issue has been fixed in v5 from 5.0.0-beta.13 onwards via https://github.com/strapi/strapi/pull/20427
